src/wasm/wasm-binary.cpp

```diff
@@ -4347,6 +4375,104 @@ void WasmBinaryBuilder::visitDrop(Drop* curr) {
   curr->finalize();
 }
 
+void WasmBinaryBuilder::visitTry(Try* curr) {
+  if (debug) {
+    std::cerr << "zz node: Try" << std::endl;
+  }
+  // For simplicity of implementation, like if scopes, we create a hidden block
+  // within each try-body and catch-body, and let branches target those inner
+  // blocks instead.
+  curr->type = getType();
+  curr->body = getBlockOrSingleton(curr->type);
+  if (lastSeparator != BinaryConsts::Catch) {
+    throwError("No catch instruction within a try scope");
+  }
+
+  // Parse instructions within (catch ...)
+  Name label = getNextLabel();
+  breakStack.push_back(
+    {label, curr->type != none && curr->type != unreachable});
+  auto start = expressionStack.size();
+  // catch instruction pushes an exnref value onto the stack, but because we
+  // model catch as not an instruction but a barrier within a try scope for
+  // readability, we assume that after catch there is an exnref value on top of
+  // the stack, and create a pop instruction that pops the exnref value from the
+  // stack. This is only for binaryen internal representation and will not be
+  // printed.
```

Comment:
What I tried to say was 'it will not be printed in binary', but I think it's confusing, because we do print it in the text format. Will remove the part after 'representation'.